### PR TITLE
Rebased PR #145 from Main

### PR DESCRIPTION
**Pull request by stuart-elliott**
- This should hopefully do the trick :)

### CODE REVIEW

**Comment by Sophist-UK** [comment]
Description from #145:

- Splitting the Package class into traits to improve readability
- Splitting the PackageServiceProvider into individual methods
- All test back to green
- Hope this helps :)

**Review by Sophist-UK — COMMENTED** [review]
In general I like the modular concepts put forward here, with the following comments:

1. I like the use of modular Traits in `Package.php`; and
2. I like the splitting of the code in the `PackageServiceProvider.php`; however
3. I would like to see the `PackageServiceProvider.php` similarly modularised into Traits; and
4. I would like to see the Test files themselves moved into matching subdirectories (moving them should all that is needed as they should still run).

I have already added the description from #145, however the PR title should be updated to describe the PR (i.e. by reusing the title from the previous PR) and not describe the historical progression.

***TL;DR:*** (For what it is worth) **I support the merging of this PR.**

I have been working on a PR of my own to:
* Add a few additional tests
* Modularise the code
* Clean up the code
* Add additional modules for:
    * Livewire views (to match the Inertia support that already exists)
    * Other things that can be loaded in a Service Provider.

but I will wait and see whether this PR is merged before working on my PR further.

**Comment by freekmurze** [comment]
Could you rebase against main to fix the conflicts?

**Comment by stuart-elliott** [comment]
> * I like the use of modular Traits in `Package.php`; and
> * I like the splitting of the code in the `PackageServiceProvider.php`; however
> * I would like to see the `PackageServiceProvider.php` similarly modularised into Traits; and

To be honest, I actually already did a part 2 for this pull request that does exactly what you mentioned, but never bothered submitting it as presumed nobody was interested. Screenshots below.

image

image

image

**Comment by stuart-elliott** [comment]
> Could you rebase against main to fix the conflicts?

Done new pull request #157 submitted with the latest updates from master.
All tests reporting green.

**Comment by Sophist-UK** [comment]
Please STOP force-pushing to Github. It makes any reviews much much more difficult.

In terms of the screen shots of code I don't like the names of the additional traits - in this context `register` and `boot` have specific meanings. Personally I think the traits should be named `processX` to match with `hasX`.
